### ISSUE TITLE
Request the running port number using port 0

### ISSUE THREAD

**Issue by gmschouten** [webpack/webpack-dev-server#998]
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behavior?**
``` javascript
server.listen(0, '127.0.0.1', () => {
 // port will automaticly assign to a free port.
});
```

**What is the expected behavior?**
``` javascript
server.listen(0, '127.0.0.1', () => {});

server.once('listening', () => {
   // while running we want to request the used port number.
   const port = server.address().port;
});

```
**If this is a feature request, what is motivation or use case for changing the behavior?**
I use webpack-dev-server to serve my application for browser-test. I want webpack to use a free port and after serving to tell my browser-test where to find the test-pages. 

**Please mention your webpack and Operating System version.**
webpack: 3.3.0
webpack-dev-server: 2.5.1

**Comment by shellscape** [comment]
We'd welcome a PR to implement this feature.

**Comment by shellscape** [comment]
this should be resolved by #1054. if it's not, please give me a ping.
